src/ipc/mem/Segment.cc

```diff
@@ -277,8 +278,7 @@ Ipc::Mem::Segment::GenerateName(const char *id)
         if (name[name.size()-1] != '/')
             name.append('/');
     } else {
-        name.append('/');
-        name.append(service_name);
+        name.append(Instance::NamePrefix("/"));
```

Comment:
@rousskov, I suspect the `if` branch of this logic needs a matching call to `Instance::NamePrefix(BasePath);` - with a related change to `BasePath` ensuring it has a trailing `/`.

Reply:
Thanks for the double-check. Accepting for now.